scene/resources/compressed_texture.cpp

```diff
@@ -339,18 +337,20 @@ Ref<Image> CompressedTexture2D::load_image_from_file(Ref<FileAccess> f, int p_si
 			} else if (data_format == DATA_FORMAT_WEBP && Image::webp_unpacker) {
 				img = Image::webp_unpacker(pv);
 			}
-
 			if (img.is_null() || img->is_empty()) {
 				ERR_FAIL_COND_V(img.is_null() || img->is_empty(), Ref<Image>());
 			}
-
-			if (first) {
-				//format will actually be the format of the first image,
-				//as it may have changed on compression
-				format = img->get_format();
-				first = false;
-			} else if (img->get_format() != format) {
-				img->convert(format); //all needs to be the same format
+			// If the image is compressed and its format doesn't match the desired format, return an empty reference.
+			// This is done to avoid recompressing the image on load.
+			ERR_FAIL_COND_V(img->is_compressed() && format != img->get_format(), Ref<Image>());
+
+			// The format will actually be the format of the header,
+			// as it may have changed on compression.
+			if (!img->is_compressed() && format != img->get_format()) {
```

Comment:
There's already a check for compressed images above.

```suggestion
			if (format != img->get_format()) {
```

Reply:
When the above check is passed, the image is either uncompressed with mismatched formats (as expected), or compressed with the same formats. For the latter, simply `format != img->get_format()` is enough to ensure that conversion of a compressed image will not happen.